fair/forcing/aerosols.py

```diff
@@ -56,3 +33,135 @@ def Stevens(emissions, E_SOx_nat=60, alpha=0.001875, beta=0.634, ref_isSO2=True)
     F = ERFari + ERFaci
     return F
 
+
+def aerocom_direct(emissions, beta = np.array(
+    [-35.29e-4, 0.0, -5.034e-4, -5.763e-4, 453e-4, -37.83e-4, -10.35e-4]),
+    scale_AR5=False):
+
+    """Calculates direct aerosol forcing based on linear relationships between
+    emissions and forcing in Aerocom models.
+
+    Reference: Myhre et al., 2013: https://www.atmos-chem-phys.net/13/1853/2013
+
+    If inputs from an RCPs SCEN file are used, the units will be correct.
+
+    Inputs: 
+        emissions: (nt x 40) emissions array
+    Keywords:
+        beta: 7-element array of forcing efficiencies in W m-2 (Mt yr-1)-1 for
+            SOx, CO, NMVOC, NOx, BC, OC, NH3 (in that order)
+        scale_AR5:       If True, scale the forcing output so that the best
+                         estimate forcing in 2011 is -0.45 W/m2 based on 2011
+                         emissions from the RCPs. The discrepancy between the
+                         Aerocom and RCP values comes from two sources; (1)
+                         there is no dust forcing considered in Aerocom/FAIR,
+                         and (2) the aerosol semi-direct is accounted for by
+                         applying this scaling factor.
+    Outputs:
+        Forcing time series
+    """
+
+    em_SOx, em_CO, em_NMVOC, em_NOx, em_BC, em_OC, em_NH3 = \
+        emissions[:,[5, 6, 7, 8, 9, 10, 11]].T
+
+    F_SOx    = beta[0] * em_SOx    * molwt.SO2 / molwt.S
+    F_CO     = beta[1] * em_CO
+    F_NMVOC  = beta[2] * em_NMVOC
+    F_NOx    = beta[3] * em_NOx    * molwt.NO / molwt.S
+    F_BC     = beta[4] * em_BC
+    F_OC     = beta[5] * em_OC
+    F_NH3    = beta[6] * em_NH3
+
+    F = F_SOx+F_CO+F_NMVOC+F_NOx+F_BC+F_OC+F_NH3
+
+    return 1.500 * scale_AR5 * F
+
+
+def ghan_indirect_emulator(emissions, fix_pre1850_RCP=True,
```

Comment:
Does the Ghan emulator slow down speed at all?

Reply:
Hi Zeb, 

On your points: Yes, yes and yes. I'll deal with the `F2x` business here mostly. The other stuff I'll promise I'll implement in the future!

The treatment of `F2x` is inconsistent at the moment. `F2x` and `tcrecs` (both specifiable by the user) are used to generate the `q` coefficients that govern the temperature change at each forcing time step. The forcing routine calculates the CO2 forcing which then feeds back into the temperature calculation (through F). 

It sounds counter-intuitive, but with the way the model interacts with `F2x` and ECS, then the lower the value of `F2x` the greater the temperature change if net non-CO2 forcing is positive. If you have a low value of `F2x` then it takes less forcing to generate the same temperature change, so anything on top of this will be more effective than under a high `F2x`.

If we allow the user to specify `F2x`, then either this makes the forcing scale factor for CO2 redundant (the first element in the `scale` option; probably a desirable outcome) or there is an inconsistency (in that the user-specified value may not agree with the number coming out of the forcing routine). CO2 forcing uncertainty comes from the carbon cycle, so I don't think we need to implement a further scaling factor on top of this. In that case I think I should adjust the forcing routine to ensure that it matches the user-specified `F2x`. This was done in v1.0 with the Myhre forcing and in v1.1 with Etminan (but incorrectly).

The Ghan method slows down the code slightly. Running just the `aerocom` method should be comparable to the old regression-based treatment as they are both linear relationships.

Reply:
Yes, I agree @chrisroadmap. I think it is important that TCR and ECS do correspond to the warming that you would get under the idealised step and ramp forcing scenarios respectively, so in combination with ```javascript F2x``` determine the feedback parameter. Different ```javascript F2x``` is physically identical to different ERF for CO2 so we should match this up in the forcing routine.